src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -274,11 +274,12 @@ case class AppDefinition(
     )
   }
 
-  private val portIndicies: Range = container.flatMap(_.hostPorts).getOrElse(portNumbers).indices
+  private val portIndices: Range = container.flatMap(_.hostPorts.filter(_.nonEmpty)).getOrElse(portNumbers).indices
```

Comment:
sounds like we need some proto changes if there's a serialization issue

On Thu, Aug 11, 2016 at 3:55 PM, Jason Gilanfarr <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/marathon/state/AppDefinition.scala
> <https://github.com/mesosphere/marathon/pull/4199#discussion_r74491617>:
>
> > @@ -274,11 +274,12 @@ case class AppDefinition(
> >      )
> >    }
> >
> > -  private val portIndicies: Range = container.flatMap(_.hostPorts).getOrElse(portNumbers).indices
> > +  private val portIndices: Range = container.flatMap(_.hostPorts.filter(_.nonEmpty)).getOrElse(portNumbers).indices
>
> if it was correctly defined as a Seq[], yes, but its not, its an
> Option[Seq[]] - so we're saying "use the container's host ports if they are
> defined and non-empty" - we can't serialize Option[Seq[]] to proto, so
> there's a major inconsistency here.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4199/files/d58766555935869e6d18f212c1bdeaa3c0cb4f8f#r74491617>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLIA-KH19TRMyVx49mvXdu4piRYzkks5qe348gaJpZM4Jga-X>
> .
>